docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
+     |  { tag: 'Pair'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<a, b> }
+
+   export const Pair = <a, b>(a: daml.Serializable<a>, b: daml.Serializable<b>): daml.Serializable<Pair<a, b>> => ({
+     decoder: () => jtv.oneOf<Pair<a, b>>(
+       jtv.object({tag: jtv.constant('Pair'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(a, b).decoder())}),
+     ),
+   });
+
+What we are interested in here is the treatment of the tuple type ``(a, b)``. As the above shows, its presence has been captured by a type ``Tuple2<a, b>`` from a package generated from the DAML standard library. Here is the definition.
+
+.. code-block:: typescript
+   :linenos:
+
+   export type Tuple2<t1, t2> = {
+     _1: t1;
+     _2: t2;
+   }
+   export const Tuple2 = <t1, t2>(t1: daml.Serializable<t1>, t2: daml.Serializable<t2>): daml.Serializable<Tuple2<t1, t2>> => ({
+     decoder: () => jtv.object({
+       _1: t1.decoder(),
+       _2: t2.decoder(),
+     }),
+   })
+
+The DAML type that the above is generated from is exactly this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Tuple2 a b =
+       Tuple2 with
+         _1 : a
+         _2 : b
+
+.. note::
+   The general rule is that DAML tuples are represented in Typescript as simple objects with fields labeled ``_1``, ``_2`` and so on.
+
+The DAML type ``Pair a b`` has been modeled in Typescript as a discriminated union with a single case. We'll see more discriminated unions in the next section.
+
+Variants
+~~~~~~~~
+
+Basic
+*****
+
+This DAML example defines a type for the language of additive expressions.
+
+.. code-block:: daml
+   :linenos:
+
+   data Expr =
+       Lit Int
+     | Var Text
+     | Add (Expr, Expr)
+
+In Typescript, the type is represented as a discriminated union. The ``Add`` branch of the type utilizes a 2-element tuple.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
+
+   export type Expr =
+     |  { tag: 'Lit'; value: daml.Int }
+     |  { tag: 'Var'; value: string }
+     |  { tag: 'Add'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<Expr, Expr> }
+
+   export const Expr:
+     daml.Serializable<Expr> & {
+     } = ({
+     decoder: () => jtv.oneOf<Expr>(
+       jtv.object({tag: jtv.constant('L'), value: jtv.lazy(() => daml.Int.decoder())}),
+       jtv.object({tag: jtv.constant('V'), value: jtv.lazy(() => daml.Text.decoder())}),
+       jtv.object({tag: jtv.constant('A'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(Expr, Expr).decoder())}),
+     ),
+   });
+
+Generic
+*******
+
+Suppose we tweak the above expression type so that it is polymorphic in the type of literals.
```

Comment:
You’ve already shown an example of polymorphism above. Does this one provide anything new to the reader?

Reply:
In the next iteration there will be no separate sub-sections for mono vs. polymorphic types (they only result in meaningful differences in the generated serialization code which will not be covered by this guide anymore).